### ISSUE TITLE
[bench request]: Add reflex (kshetrajna12/reflex-qwen3.5-4b-lora, TypeSafe wire format)

### ISSUE THREAD

**Issue by kshetrajna12** [fstandhartinger/jevbench#3]
Request to add [reflex](https://github.com/kshetrajna12/reflex) (Qwen3.5-4B + a LoRA adapter,
[kshetrajna12/reflex-qwen3.5-4b-lora](https://huggingface.co/kshetrajna12/reflex-qwen3.5-4b-lora);
MIT code and adapter, Apache-2.0 base weights) to the ranked systems.

**What it is.** An open re-creation of the Jev design: the state is encoded once and cached,
every question runs as an isolated branch in one pass, and the answer is read directly from the
label logits (no decoding). Trained with a proper scoring rule on public datasets and synthetic
correct-by-construction data ("RLCD-lite"); temperatures fitted per primitive on our own held-out
set. It serves TypeSafe's wire format, so your `typesafe` adapter works unchanged.

```sh
git clone https://github.com/kshetrajna12/reflex && cd reflex && git checkout 1add693
uv sync
REFLEX_API_KEY=any uv run reflex-serve --model Qwen/Qwen3.5-4B \
    --adapter kshetrajna12/reflex-qwen3.5-4b-lora --served-name reflex --host 0.0.0.0 --port 8000
# adapter + calibration.json are pulled from the Hub; ~9 GB bf16, any 16 GB CUDA GPU;
# the first request compiles Triton kernels (~20 s), so send one warm-up request before timing.

export REFLEX_KEY=any
python -m jevbench.cli run --tasks datasets/public/original.jsonl \
  --adapter typesafe --endpoint http://127.0.0.1:8000 --key-env REFLEX_KEY --model reflex \
  --cost-basis self_hosted_gpu --reserve-usd 0 ...
```
A Dockerfile (`docker/Dockerfile`) and the full recipe are in `docs/SERVING.md`.

**Our own numbers on the public items** (your adapter's requests, one at a time, argmax over the
exact label set):

| tier | items | reflex |
|---|---|---|
| easy | 48 | 1.000 |
| standard (original) | 72 | 0.944 |
| hard | 111 | 0.604 |

Hard-tier top-label ECE 0.117; distribution fidelity on the 10 public probability items 0.689.
We are not claiming a rank from this: the held-out and imported items are yours, and speed and
cost are measured from your server.

**Disclosure.** The training recipes were designed after looking at which public families the
raw model failed (answer adequacy, rule-based routing, base-rate probabilities; a long-policy
recipe exists in the repo but this adapter was not trained on it). No benchmark item or paraphrase was used in training; `reflex-data check-overlap` refuses to
train if any training row shares an 8-word sequence with a public item. The public 231 were used
as a development gate four times (three adapters, one re-calibration), so expect the held-out hard items to score somewhat below the
public ones.

Happy to answer questions here or rerun anything.

**Comment by kshetrajna12** [comment]
Update to this request: please evaluate the **frozen base model with reflex's default
prompt and no adapter** instead of the LoRA adapter named above. Our controls since
filing showed the adapter traded accuracy on long, ambiguous items for accuracy on items
that resemble its training data, and that two prompt-level changes (a lettered yes/no
readout and "Evidence / Criterion" framing), selected on held-out external sets, did
better without touching a weight. Details and every number are in
`docs/results/frozen-vs-trained.md` in the repo.

Configuration:

```sh
git clone https://github.com/kshetrajna12/reflex && cd reflex && git checkout dae6799
uv sync
REFLEX_API_KEY=any uv run reflex-serve --model Qwen/Qwen3.5-4B --served-name reflex --host 0.0.0.0 --port 8000
```

No adapter, no calibration file: `Qwen/Qwen3.5-4B` in bf16, reflex's default prompt at
that commit, probabilities read from the option-label logits with temperature 1. Same
`/v1/systemone` wire format as before; your `typesafe` adapter runs unchanged. Any 16 GB
CUDA GPU; the first request compiles Triton kernels (~20 s), so send one warm-up request.

Our own numbers on the public items (your adapter's requests, one at a time):

| tier | items | frozen + reflex prompt | previously filed adapter |
|---|---|---|---|
| easy | 48 | 1.000 | 1.000 |
| standard (original) | 72 | 0.917 | 0.944 |
| hard | 111 | 0.658 | 0.604 |

Hard-tier top-label ECE 0.086, fidelity on the 10 public probability items 0.708. Not
claiming a rank from this. Disclosure as before: the public items were used as a
development gate (now nine self-runs across adapters, prompts and calibration variants);
prompt changes were selected on four external labelled sets first and confirmed on the
public items second; no benchmark item was used for training or for fitting anything.

If you would rather keep the adapter row as filed, that is fine too; the frozen
configuration is the one we will stand behind going forward.

**Comment by fstandhartinger** [comment]
Thanks — this is logged for reachability checks and the JevBench measurement queue. We’ll only make benchmark claims after our own run.

- Harold

**Comment by fstandhartinger** [comment]
reflex (Qwen3.5-4B + your LoRA and calibration file) ran the full frozen v1.2 set — 534 decisions including the
held-out ones — through `reflex-serve` at 1add693 on our RunPod H100, one request at a time, unchanged `typesafe` adapter.

**#5 of 36, JevBench Score 71.7** (Intelligence 86.7, Calibration 75.2, Speed 68.0, Cost 59.7). Accuracy: easy 100.0 %, standard 94.8 %, judge 97.3 %, hard 63.2 %. Latency p50 1.80 s raw, 3.75 s after the standard self-host adjustment (x2 + 0.15 s); our pod was in Canada, so the raw figure includes the trip from our server in Germany; cost $0.0221 per 1,000 decisions.

It is the best-placed newcomer this round. One run was thrown away and redone before any of this: another server on the
same pod was still compiling during our first attempt, which slowed your latencies; that run is kept with its reason.

Page: https://benchmarkheaven.com/jev-models · data and method: https://github.com/fstandhartinger/jevbench/tree/v1.2.8

- Harold

**Comment by kshetrajna12** [comment]
Thank you, Harold, and for redoing the run when the pod was contended.

One note for the next round: this run used commit `1add693` (the LoRA adapter and calibration file from the original filing). The configuration we now ship, tag `stable` at `19586a1`, is the frozen Qwen3.5-4B with two option orders and no adapter or calibration file; on the public items it scores hard 0.685 vs 0.604 and hard-tier ECE 0.081 vs 0.117, at about 1.1x the latency:

```sh
git clone https://github.com/kshetrajna12/reflex && cd reflex && git checkout stable
uv sync
REFLEX_API_KEY=any uv run reflex-serve --stable --served-name reflex --host 0.0.0.0 --port 8000
```

If a rerun of `reflex` at that tag fits a future round, we'd be grateful. Should that be a new bench-request issue, or is a comment here the right place? No change requested for reflex-27b.

**Comment by kshetrajna12** [comment]
A follow-up on latency, since it decides most of the Speed axis. We measured the same server on an H100: `reflex-serve --stable` answers a single decision in under 100 ms server-side (the response header `x-reflex-latency-ms` reports it per request), and on our own DGX Spark a hard-tier item is ~0.18 s. The 1.80 s p50 in the run must have been the path to the pod rather than the model: both our entries measured within 5 % of each other (1.80 s for the 4B, 1.89 s for the 27B) despite a 7x difference in model size, which is what a fixed per-request cost looks like.

So for the rerun of `reflex` at tag `stable` (commit `19586a1`), two practical asks, both within your existing protocol as I read the SemIf entry's notes: a pod in an EU region, and the pod's direct TCP port rather than a proxied HTTP endpoint. If it helps, `x-reflex-latency-ms` lets you split server time from transport in your logs; a startup diagnostic that names the kernel path in use is landing on `main` this week but is not needed for the run.

Configuration is unchanged from my previous comment:

```sh
git clone https://github.com/kshetrajna12/reflex && cd reflex && git checkout stable
uv sync
REFLEX_API_KEY=any uv run reflex-serve --stable --served-name reflex --host 0.0.0.0 --port 8000
```

Happy to open a fresh bench-request issue for it if that is your preferred flow.

**Comment by fstandhartinger** [comment]
Congratulations on the JevBench v1.4 release, kshetrajna12! Thanks for building and sharing your system.

- reflex 4B: #5 at 53.99
- reflex-27b: #51 at 17.84

reflex 4B was #5 in v1.3; reflex-27b was #17 in v1.3. V1.4 blends in 308 fresh sealed decisions, uses an equal-weight harmonic mean, and applies separate Speed and Cost gates. For your leading row, sealed accuracy was 28.2% versus 79.2% on the public slice. The >25-point gap adjustment applied; the below-50 Intelligence gate also applied.

Full board and method: https://benchmarkheaven.com/jev-models
